### DISCUSSION TITLE
WeConnect for NodeRed?

### DISCUSSION THREAD

**Discussion by wizzywik** [tillsteinbach/WeConnect-mqtt#68] [Ideas]
Any plans for a package for NodeRed? That's my favorite integration platform.

**Comment by tillsteinbach** [comment]
That is based on java script, right? I guess there is a more realistic chance to ask the people here: https://github.com/nightsha-de/npm-vwconnectapi
Apart from that: If you can call scripts from NodeRed you can also execute all commands with WeConnect-cli instead of MQTT: https://github.com/tillsteinbach/WeConnect-cli

**Comment by wizzywik** [comment]
Thnx Till, I will ask nightsha. Saw this before but don't know how to install/implement in NodeRed. 
A bit the same for your cli. Don't know how to do that. If I see some examples or tutorials it might help, but haven't found yet. For now I will use the MQTT.

**Reply by wizzywik** [reply]
Hi Till, MQTT package working! Great. 
Implemented controls/charging and controls/climitisation too. Sometimes they work, but most off the time they don't. Is that a problem with the VW servers? Checked with a MQTT sniffer, and commands look good.
And finally, is there an overview off all (relevant) controls?
Thnx again!

**Reply by wizzywik** [reply]
Additional: a restart of the docker container solves the problem temporarily. After restart, my commands(controls) work well.

**Reply by tillsteinbach** [reply]
What model of car do you have? I also have problems with my Passat right now. The ID though works great. It seems like there are problems bridging the commands to the legacy Cars

**Reply by wizzywik** [reply]
ID4. Your packages reports "weconnect/0/vehicles/WVGXXXX/controls/charging none" just after a restart. Then my start/stop commands work well. After some time or events (have to check that in more detail) the controls don't work anymore. Still data is coming in at the 300 sec interval.

**Reply by tillsteinbach** [reply]
does it set an error in the error topic? There are some errors that can happen, e.g. if you send too many commands you have to go to the car and drive a round...

**Reply by tillsteinbach** [reply]
I might have found a problem. Can you try if it works if you do a 'stop' and afterwards a 'start'. There is a bug that does not accept a start if there was no stop before. I will fix that as soon as possible

**Reply by wizzywik** [reply]
When I run in the same problem again, I will try to do some reproduction tests (start/stop/start and so on).

**Reply by wizzywik** [reply]
Can I use a 'weconnectForceUpdate' within the minimal refresh interval (300s)? For example to force an update at the start of a fast DC charging session? I saw that IdDataLogger has a automatic adjustment of the refresh interval when charging starts. Then the interval will drop significantly below 300 ("base-updaterate": 600, "increased-updaterate": 60).

**Reply by tillsteinbach** [reply]
Force update should always work. I‘m currently thinking of ways to dynamically change the interval. I don’t want to put too much load on the VW servers to not risk that they lock us out.
